### PR TITLE
Add guidelines and README template for contributing to examples

### PR DESCRIPTION
**Pull request by bermudezmt**
This should hopefully standardize our tutorials.

Partially addresses #371

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+```
+|____README.md
+|____www
+|____images
+| |____deploy.gif
+| |____deploy.cast
+|____Pulumi.yaml
+|____package.json
+|____tsconfig.json
+|____index.ts
```
Comment (mikhailshilkov) [inline CONTRIBUTING.md]: Why do you have underscores in front of the file names?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- Overview paragraph
+- Prerequisites
+- Deploy the App
+   - Step 1 — Doing the First Thing
+   - Step 2 — Doing the Next Thing
+   …
+   - Step n — Doing the Last Thing
+- Clean Up
+- Summary
+- Next Steps (Optional)
```
Comment (mikhailshilkov) [inline CONTRIBUTING.md]: Should we explicitly admit that most of our existing samples don't really follow this structure and why?

File: example-readme-template.md.txt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+## Next Steps
+
+<!-- Optionally include an unordered list of relevant Pulumi tutorials. -->
+
+<!-- Example:
+- [Create a load-balanced, hosted NGINX container service](https://www.pulumi.com/docs/tutorials/aws/ecs-fargate/)
+- [Create an EC2-based WebServer and associated infrastructure](https://www.pulumi.com/docs/tutorials/aws/ec2-webserver/)
+-->
+
```
Comment (mikhailshilkov) [inline example-readme-template.md.txt]: Not sure where, but I would add a link to a reference README file that we think is the best example.

File: CONTRIBUTING.md (hunk 1 above)
Comment (bermudezmt) [reply CONTRIBUTING.md]: This is the output of `tree`. I've always liked using it for showing a directory structure.

File: CONTRIBUTING.md (hunk 2 above)
Comment (bermudezmt) [reply CONTRIBUTING.md]: Good call.

File: example-readme-template.md.txt (hunk 3 above)
Comment (bermudezmt) [reply example-readme-template.md.txt]: I would do this as a follow-up change after editing one of the featured tutorials: Now tracked as an item in https://github.com/pulumi/examples/issues/371.

**Pull request merged**